lib/internal/repl.js

```diff
@@ -167,6 +174,8 @@ function setupHistory(repl, historyPath, oldHistoryPath, ready) {
                             'REPL session history will not be persisted.\n');
       }
       if (!threw) {
+        _printHistoryDeprecation(oldHistoryPath);
```

Comment:
I think this should be called in `setupHistory()` iff `oldHistoryPath` is not `null` or `undefined`. By calling it here, it won't be displayed for users who have set the `NODE_REPL_HISTORY_FILE` env variable, but opening or parsing that history file throws an `Error`.

Reply:
Hmm, fair point. But wouldn't we want to only show the error message to NODE_REPL_HISTORY_FILEs that actually parse correctly? If the file is broken, they probably wouldn't be able to convert it to the new file format anyway, right? 😬

Reply:
Hm - ping @lance @maclover7 ?

My guess would be that there's nobody left using the old format anyway by now and that this particular difference would really not matter anyway

Reply:
Sorry for my latency. @addaleax @maclover7 your points are valid. TBH, I think it would be more correct if moved to `setupHistory()`, but I won't block the PR on that.